demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -0,0 +1,1046 @@
+     * as the callback context which is used by the callback to send a task
+     * notification to this task.*/
+    mqttSubscription[ 0 ].qos = MQTTQoS0;
+    mqttSubscription[ 0 ].pTopicFilter = MQTT_EXAMPLE_TOPIC;
+    mqttSubscription[ 0 ].topicFilterLength = ( uint16_t ) strlen( MQTT_EXAMPLE_TOPIC );
+
+    result = MQTT_GetUnsubscribePacketSize( mqttSubscription,
+                                            sizeof( mqttSubscription ) / sizeof( MQTTSubscribeInfo_t ),
+                                            &remainingLength,
+                                            &packetSize );
+    assert( result == MQTTSuccess );
+    /* Make sure the packet size is less than static buffer size */
+    assert( packetSize < pFixedBuffer->size );
+
+    /* Get next unique packet identifier */
+    unsubscribePacketIdentifier = getNextPacketIdentifier();
+
+    result = MQTT_SerializeUnsubscribe( mqttSubscription,
+                                        sizeof( mqttSubscription ) / sizeof( MQTTSubscribeInfo_t ),
+                                        unsubscribePacketIdentifier,
+                                        remainingLength,
+                                        pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send Unsubscribe request to the broker. */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+/*-----------------------------------------------------------*/
+
+static void mqttKeepAlive( int tcpSocket,
+                           MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    int status;
+    size_t packetSize = 0;
+
+    status = MQTT_GetPingreqPacketSize( &packetSize );
+
+    /* PingReq is fixed length packet, therefore there is no need to calculate the size,
+     * just makes sure static buffer can accommodate ping request. */
+    assert( packetSize <= SHARED_BUFFER_SIZE );
+
+    result = MQTT_SerializePingreq( pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send Ping Request to the broker. */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+
+/*-----------------------------------------------------------*/
+
+static void mqttDisconnect( int tcpSocket,
+                            MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    int32_t status;
+    size_t packetSize = 0;
+
+    status = MQTT_GetDisconnectPacketSize( &packetSize );
+
+    assert( packetSize <= pFixedBuffer->size );
+
+    result = MQTT_SerializeDisconnect( pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send disconnect packet to the broker */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
```

Comment:
This documentation is already present at the top of this file and can be removed from here.  Same for other functions.